lxd/storage/backend_lxd.go

```diff
@@ -0,0 +1,418 @@
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceFromBackup(i Instance, sourcePath string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceFromCopy(i Instance, src Instance, snapshots bool, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceFromImage(i Instance, fingerprint string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceFromMigration(i Instance, conn *websocket.Conn, args migration.SinkArgs, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RenameInstance(i Instance, newName string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteInstance(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MigrateInstance(i Instance, snapshots bool, args migration.SourceArgs) (migration.StorageSourceDriver, error) {
+	return nil, ErrNotImplemented
+}
+
+func (b *lxdBackend) RefreshInstance(i Instance, src Instance, snapshots bool, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) BackupInstance(i Instance, targetPath string, optimized bool, snapshots bool, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) GetInstanceUsage(i Instance) (uint64, error) {
+	return 0, ErrNotImplemented
+}
+
+func (b *lxdBackend) SetInstanceQuota(i Instance, quota uint64) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MountInstance(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) UnmountInstance(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) GetInstanceDisk(i Instance) (string, string, error) {
+	return "", "", ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceSnapshot(i Instance, name string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RenameInstanceSnapshot(i Instance, newName string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteInstanceSnapshot(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RestoreInstanceSnapshot(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MountInstanceSnapshot(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) UnmountInstanceSnapshot(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateImage(img api.Image, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+// DeleteImage removes an image from the database and underlying storage device if needed.
+func (b *lxdBackend) DeleteImage(fingerprint string, op *operations.Operation) error {
```

Comment:
@stgraber I've added DeleteImage now, although in the current dir implementation I cant see that it actually does anything except remove the DB entry. Is that expected because it doesn't need to create an image volume?

Reply:
There shouldn't even be a DB entry in the dir case.

Anything which has OptimizedImages() return false shouldn't have image type volumes at all.

Reply:
@stgraber I see thanks. I guess there is no harm in trying to remove it though. But that does explain why in my tests on the current implementation I couldnt see that row being created.